presto-main/src/main/java/com/facebook/presto/sql/Serialization.java

```diff
@@ -112,11 +116,10 @@ public VariableReferenceExpressionDeserializer(TypeManager typeManager)
 
         @Override
         public Object deserializeKey(String key, DeserializationContext ctxt)
-                throws IOException
         {
-            int p = key.indexOf("(");
-            if (p <= 0 || key.charAt(key.length() - 1) != ')') {
-                throw new IllegalArgumentException(format("Expect key to be of format 'name(type)', found %s", key));
+            int p = key.indexOf(VARIABLE_TYPE_OPEN_BRACKET);
+            if (p <= 0 || key.charAt(key.length() - 1) != VARIABLE_TYPE_CLOSE_BRACKET) {
```

Comment:
This format has nothing to do with JSON :D . We are just using Jackson for convenience :)

Reply:
It's not completely true. This is needed because JSON key has to be simple types and `VariableReferenceExpression` is not. So we need to serialize it to a string instead.